pandas/core/arrays/arrow/array.py

```diff
@@ -1483,6 +1483,10 @@ def to_numpy(
     def map(self, mapper, na_action: Literal["ignore"] | None = None):
         if is_numeric_dtype(self.dtype):
             return map_array(self.to_numpy(), mapper, na_action=na_action)
+        elif self.dtype == "timestamp[ns][pyarrow]":
```

Comment:
Instead of adding an `elif` you can modify the existing `if` statement as `if is_numeric_dtype(self.dtype) or self.dtype.kind in "mM":`